_posts/2017/2017-02-20-continuous-documentation-with-ansible.md

```diff
@@ -0,0 +1,207 @@
+Motivation for doing it like this is that the exact same playbooks can be used for different environments / inventories (see below).
+
+## Roles
+While playbooks are the entry point, one will soon be looking at [Roles](http://docs.ansible.com/ansible/playbooks_roles.html), because playbooks should not contain single tasks, but only sets of roles (that are applied to groups of hosts).  
+When learning about a role, the best place to start is `defaults/main.yml`.  
+This file is supposed to contain all variables that are being used in the role, and - as the filename implies - their default values. This file serves the primary purpose of documenting the role.  
+When using a role written by others, these variables are meant to be used to adapt the role to specific needs.
+
+Example `defaults/main.yml` of `epages-unity` role:  
+{% highlight yml %}
+epages_unity_folder: /srv/epages/unity
+epages_unity_use_docker_compose: yes
+epages_unity_use_systemd: "{{ not epages_unity_use_docker_compose }}"
+
+epages_unity_epages6_appserver_range: "10045-10048"
+epages_unity_web_server_port: 7000
+epages_unity_ssl_web_server_port: "{{ epages_unity_web_server_port }}"
+epages_unity_url: "{{ ansible_fqdn }}"
+epages_unity_api_url: "{{ epages_unity_url }}/api/v2"
+
+epages_unity_create_shop: yes
+epages_unity_shop_name: demounity
+{% endhighlight %}
+At the top there is a directory, which probably refers to the host.
+Since the role requires Docker (see playbook example), and there is a reference to docker-compose in the next property, one could (correctly) assume that the file being copied there could be a docker-compose definition file.
+However there is also a property `epages_unity_use_systemd` which is apparently mutually exclusive with `epages_unity_use_docker_compose`.
+The purpose of having two different ways of starting the application could (and should) be questioned, it really just documents the circumstance, that while either of the two ways should work, one or the other might be broken while development is still ongoing.  
+But as Agile thinking suggests this should not prevent us from [automating the deployment as early as possible](http://www.theserverside.com/tip/Try-an-Agile-deployment-strategy).
+
+The file goes on with some more configuration of what seem to be application properties, and finally two options to create a shop with a given name when deploying.
+By looking at the defaults, anyone could get a good idea about the scope of the role without looking at the [tasks](http://docs.ansible.com/ansible/playbooks_intro.html#tasks-list).  
+As mentioned before, roles do not necessarily need to live in the same repository as playbooks, because Ansible supports [role dependencies](http://docs.ansible.com/ansible/playbooks_roles.html#role-dependencies).
+
+## Variables
```

Comment:
Maybe this could be a good point for starting the second post.
Ending the first one with something like "In a follow up post, I will discuss Variables...." whatever. Just to have a smooth ending and a entry point for the next one.